packages/Search/test/tstLinearBVH.cpp

```diff
@@ -41,134 +41,289 @@ void check_results( DataTransferKit::BVH<DeviceType> &bvh,
-    int const n = 2;
-    Kokkos::View<DataTransferKit::Box *, DeviceType> boxes( "boxes", n );
-    FillBoxes<DeviceType> fill_boxes_functor( boxes );
-    Kokkos::parallel_for( "file_boxes_functor",
-                          Kokkos::RangePolicy<ExecutionSpace>( 0, n ),
-                          fill_boxes_functor );
-    Kokkos::fence();
+    // NOTE: `points` is not a very descriptive name here. It stores both the
+    // actual point and the number k of neighbors to query for.
+    int const n = points.size();
+    Kokkos::View<DataTransferKit::Details::Nearest *, DeviceType> queries(
+        "nearest_queries", n );
+    auto queries_host = Kokkos::create_mirror_view( queries );
+    for ( int i = 0; i < n; ++i )
+        queries_host( i ) = DataTransferKit::Details::nearest(
+            points[i].first, points[i].second );
+    Kokkos::deep_copy( queries, queries_host );
+    return queries;
+}
 
-    DataTransferKit::BVH<DeviceType> bvh( boxes );
-    auto do_nothing_1 = KOKKOS_LAMBDA( int ){};
-    auto do_nothing_2 = KOKKOS_LAMBDA( int, double ){};
-    DataTransferKit::Point p1 = {{0., 0., 0.}};
-    details::TreeTraversal<DeviceType>::query( bvh, details::nearest( p1, 1 ),
-                                               do_nothing_2 );
-
-    details::TreeTraversal<DeviceType>::query( bvh, details::within( p1, 0.5 ),
-                                               do_nothing_1 );
+template <typename DeviceType>
+Kokkos::View<DataTransferKit::Details::Within *, DeviceType>
+make_within_queries(
+    std::vector<std::pair<DataTransferKit::Point, double>> const &points )
+{
+    // NOTE: `points` is not a very descriptive name here. It stores both the
+    // actual point and the radius for the search around that point.
+    int const n = points.size();
+    Kokkos::View<DataTransferKit::Details::Within *, DeviceType> queries(
+        "within_queries", n );
+    auto queries_host = Kokkos::create_mirror_view( queries );
+    for ( int i = 0; i < n; ++i )
+        queries_host( i ) = DataTransferKit::Details::within(
+            points[i].first, points[i].second );
+    Kokkos::deep_copy( queries, queries_host );
+    return queries;
 }
 
 TEUCHOS_UNIT_TEST_TEMPLATE_1_DECL( LinearBVH, bounds, DeviceType )
 {
-    Kokkos::View<DataTransferKit::Box *, DeviceType> boxes( "boxes", 2 );
-    auto boxes_host = Kokkos::create_mirror_view( boxes );
-    boxes_host( 0 ) = DataTransferKit::Box( {2., 2., 4., 4., 6., 6.} );
-    boxes_host( 1 ) = DataTransferKit::Box( {1., 1., 3., 3., 5., 5.} );
-    Kokkos::deep_copy( boxes, boxes_host );
-    DataTransferKit::BVH<DeviceType> bvh( boxes );
-    auto bounds = bvh.bounds();
-    TEST_EQUALITY( bounds[0], 1.0 );
-    TEST_EQUALITY( bounds[1], 2.0 );
-    TEST_EQUALITY( bounds[2], 3.0 );
-    TEST_EQUALITY( bounds[3], 4.0 );
-    TEST_EQUALITY( bounds[4], 5.0 );
-    TEST_EQUALITY( bounds[5], 6.0 );
+    // auto bvh = make_bvh<DeviceType>( {
```

Comment:
Why is this test commented?

Reply:
Forgot to erase it.

Reply:
@Rombur I realize now I must have forgotten to cherry pick one of the commits.  Will update later tonight...